src/main/java/com/zaxxer/nuprocess/windows/WindowsProcess.java

```diff
@@ -777,7 +780,12 @@ public int compare(Map.Entry<String, String> e1, Map.Entry<String, String> e2)
       {
          this.pipeHandle = pipeHandle;
          this.ioCompletionKey = ioCompletionKey;
-         this.overlapped = new OVERLAPPED();
+
+         // The OVERLAPPED structure is required to make non-blocking ReadFile and WriteFile calls,
+         // but its state is never read from nor written to in Java, so we disable auto-sync so JNA
+         // doesn't waste time marshaling contents to/from native memory
+         overlapped = new OVERLAPPED();
+         overlapped.setAutoSynch(false);
```

Comment:
In my testing, which involved a process that took ~122,000 `ReadFile` calls to complete, disabling auto-sync on `OVERLAPPED` cut ~20 seconds off the overall execution time.